{"priority": "minor", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/osrf/gazebo.json"}, "html": {"href": "#!/osrf/gazebo"}, "avatar": {"href": "data/bytebucket.org/ravatar/{51a0cd5d-8697-4eb1-8b08-e919ee881e1c}ts=1694483"}}, "type": "repository", "name": "gazebo", "full_name": "osrf/gazebo", "uuid": "{51a0cd5d-8697-4eb1-8b08-e919ee881e1c}"}, "links": {"attachments": {"href": "data/repositories/osrf/gazebo/issues/1179/attachments_page=1.json"}, "self": {"href": "data/repositories/osrf/gazebo/issues/1179.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/gazebo/issues/1179/watch"}, "comments": {"href": "data/repositories/osrf/gazebo/issues/1179/comments_page=1.json"}, "html": {"href": "#!/osrf/gazebo/issues/1179/exceptions-no-longer-print-file-line-from"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/gazebo/issues/1179/vote"}}, "reporter": {"display_name": "Steve Peters", "uuid": "{2ccfed09-18b8-4921-8d58-15ef01092802}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D"}, "html": {"href": "https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/1fb4816dad9e68337d3096f750951f6cd=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsSP-1.png"}}, "nickname": "Steven Peters", "type": "user", "account_id": "557058:5de38267-b118-494c-aa76-4fab35448816"}, "title": "Exceptions no longer print file:line from where they were thrown", "component": {"name": "common", "links": {"self": {"href": "data/repositories/osrf/gazebo/components/138866.json"}}}, "votes": 0, "watches": 2, "content": {"raw": "Prior to pull request #894, a `gzthrow` would print the file and line number in a console message, for example:\r\n~~~\r\n79: Exception [ODEJoint.cc:231] ODEHinge2Joint cannot be connected to the world\r\n~~~\r\nThe switch to using `gzerr` in `Exception::Print()` means that `Exception.cc:52` is printed with every exception, which is not useful.\r\n~~~\r\n[Err] [Exception.cc:52] EXCEPTION: Unable to start server[bind: Address already in use]. There is probably another Gazebo process running.\r\n~~~\r\nWe should use `std::cerr` instead of `gzerr` or figure out a way to use the correct file and line number.", "markup": "markdown", "html": "<p>Prior to <a href=\"#!/osrf/gazebo/pull-requests/894/improve-console-output-fix-for-issue-794\" rel=\"nofollow\" class=\"ap-connect-link\">pull request #894</a>, a <code>gzthrow</code> would print the file and line number in a console message, for example:</p>\n<div class=\"codehilite\"><pre><span></span><span class=\"mi\">79</span><span class=\"o\">:</span> <span class=\"n\">Exception</span> <span class=\"o\">[</span><span class=\"n\">ODEJoint</span><span class=\"o\">.</span><span class=\"na\">cc</span><span class=\"o\">:</span><span class=\"mi\">231</span><span class=\"o\">]</span> <span class=\"n\">ODEHinge2Joint</span> <span class=\"n\">cannot</span> <span class=\"n\">be</span> <span class=\"n\">connected</span> <span class=\"n\">to</span> <span class=\"n\">the</span> <span class=\"n\">world</span>\n</pre></div>\n\n\n<p>The switch to using <code>gzerr</code> in <code>Exception::Print()</code> means that <code>Exception.cc:52</code> is printed with every exception, which is not useful.</p>\n<div class=\"codehilite\"><pre><span></span>[Err] [Exception.cc:52] EXCEPTION: Unable to start server[bind: Address already in use]. There is probably another Gazebo process running.\n</pre></div>\n\n\n<p>We should use <code>std::cerr</code> instead of <code>gzerr</code> or figure out a way to use the correct file and line number.</p>", "type": "rendered"}, "assignee": null, "state": "closed", "version": {"name": "all", "links": {"self": {"href": "data/repositories/osrf/gazebo/versions/225575.json"}}}, "edited_on": null, "created_on": "2014-05-06T18:31:45.743239+00:00", "milestone": null, "updated_on": "2016-09-14T19:45:14.647186+00:00", "type": "issue", "id": 1179}